shared/src/pool_fetching.rs

```diff
@@ -189,12 +191,13 @@ impl CachedPoolFetcher {
         let cache_results: Vec<_> = cache_hits
             .iter()
             .filter_map(|pair| cache.1.get(pair))
+            .flatten()
             .cloned()
             .collect();
 
         let mut inner_results = self.inner.fetch(cache_misses).await;
         for miss in &inner_results {
-            cache.1.insert(miss.tokens, miss.clone());
+            cache.1.entry(miss.tokens).or_default().push(miss.clone());
```

Comment:
What does the term "miss" represent here?

Reply:
could we include a comment here on what is happening. I find this very cryptic and I am not sure I even understand the difference between the changes.

Reply:
> What does the term "miss" represent here?

It represents a cache miss (the opposite of a cache hit).

> I am not sure I even understand the difference between the changes.

It's exactly the same as [in the other fix](https://github.com/gnosis/gp-v2-services/pull/633/files#diff-f6933a68a986a710ace700404bf73cb176ec2a1abac83803335794b8f91db01bR324)

Instead of overriding the hash map with a single value, we load the existing value (which is a list) and append to it. Happy for a suggestion on how to write this in a way that is easier to understand, however I'm not sure a comment is a good idea (comments should focus on why something is happening and not repeat the "what" which is already implicit by the code).

Reply:
That is already helpful enough. Thanks.